Show a general error message when the phone number verification request is failed

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This PR solves a task of 📌 [Clarify API error messages](https://github.com/woocommerce/google-listings-and-ads/issues/1993#tasks-clarify-api-errors) in #1993

Closes #1101 

This PR fixes the blank error message when requesting verification with an invalid phone number. Please also refer to #1998 for why this PR can only show a general error message.

### Screenshots:

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/b907b2ab-1222-4810-bcc1-4500620130b4)

### Detailed test instructions:

1. Go to the phone number editing page under the settings page or step 3 of the onboarding flow.
2. Request verification with an invalid phone number. For example, `+31` and `111111111`.
3. Check if the UI shows a proper error message.

### Additional details:

I didn't add tests in this PR as I hope to add more comprehensive tests for the `VerifyPhoneNumberContent` component before the end of the project if there is still time. Added a follow-up item to 📌 [Add tests](https://github.com/woocommerce/google-listings-and-ads/issues/1993#fe-tests).

### Changelog entry

> Fix - Show a general error message when the phone number verification request is failed.

eason9487 committed:
Show a general error message when the phone number verification request is failed

jorgemd24 commented on js/src/data/actions.js:
```diff
@@ -610,6 +610,21 @@ export function* requestPhoneVerificationCode( country, phoneNumber, method ) {
 			verificationId: response.verification_id,
 		};
 	} catch ( error ) {
+		// TODO: Remove the handling of 'backendError' and add the case for 'badRequest'.
+		//
+		// Currently, 'badRequest' won't be presented but maybe someday it can be handled.
+		// Ref:
+		// - https://github.com/woocommerce/google-listings-and-ads/issues/1101
+		// - https://github.com/woocommerce/google-listings-and-ads/issues/1998
```
I'm unsure whether this should be marked as a "TODO" or considered pending since we lack knowledge about potential changes in the API response down the line. One option could be to omit the "TODO" label and simply leave the comments as they are.

eason9487 replied:
Removing only TODO might make the semantics deviate somewhat from what this error handling was supposed to do. I rephrased it in https://github.com/woocommerce/google-listings-and-ads/pull/1999/commits/cbda28317dd7547b4d6fc7fed62a6b4aa263045c to only describe the "why" but didn't specify "what" to do.

jorgemd24 replied:
Thanks @eason9487 for the adjustment, I think the comment is clearer now.

jorgemd24 reviewed (approved):
Thanks @eason9487 for addressing this. I left a minor comment about the "TODO" label. LGTM 👍

eason9487 committed:
Rephrase the code comment of error hanlding in the requestPhoneVerificationCode action.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/1999#discussion_r1240183749
